### ISSUE TITLE
Error in Designer after upgrade to version 2.4.1

### ISSUE THREAD

**Issue by akghasemi** [elsa-workflows/elsa-core#2686]
Hi,
I have upgraded Elsa version 2.2 to 2.4.1 but now designer has errors , I have cleared browser cache as well but it didn't help.
when I click on create Workflow button I get some javascript error

![image](https://user-images.githubusercontent.com/46082076/150556962-08975ff6-867a-4c29-ae8d-d1fe728beb3b.png)

what can be the reason?

**Comment by sfmskywalker** [comment]
Looks like the dashboard is trying to render the workflow test panel. But unless you enabled workflow testing in Startup, this should not happen. Clearly a bug.

**Comment by akghasemi** [comment]
Thanks for your reply,
is there any work around that I can fix this temporarily?

**Comment by sfmskywalker** [comment]
Can't say, I have yet to look into it and reproduce this issue. If you're pressed for time, then maybe better to go back to a previous version.

**Comment by sfmskywalker** [comment]
I can't seem to reproduce this issue. If you can give me access to your project or send me a reproduction then I will take a look.

**Comment by akghasemi** [comment]
unfortunately the project is too big but I can give you my startup, maybe this help
 services.AddElsa(elsa =>
                {
                    var elsaSection = Configuration.GetSection("Elsa");
                    ElsaCustomContext.SetSchemaName(elsaSection.GetValue<string>("Schema"), databaseSetting.DatabaseType);

                    if (databaseSetting.DatabaseType == enmDatabaseType.Oracle)
                    {
                        elsa.UseEntityFrameworkPersistence<ElsaCustomContext>(options => options
                          .UseOracle(Configuration.GetConnectionString("DefaultConnectionORA"), b => b.UseOracleSQLCompatibility(databaseSetting.CompatiblitiyVersion)), false);
                    }
                    else
                    {
                        elsa.UseEntityFrameworkPersistence<ElsaCustomContext>(options => options
                          .UseNpgsql(Configuration.GetConnectionString("DefaultConnectionPG")), false);
                    }

                    elsa.AddHttpActivities(elsaSection.GetSection("Server").Bind);
                    elsa.AddQuartzTemporalActivities();
                    elsa.AddConsoleActivities();
                    elsa.AddFileActivities();

                    elsa.AddEmailActivities(elsaSection.GetSection("SMTP").Bind);
                    elsa.AddWorkflowsFrom<Startup>();

                    elsa.AddActivity<RegisterApprovalActivity>();
                    elsa.AddActivity<CheckApprovalActivity>();
                    elsa.AddActivity<ChangeApprovalItemStateActivity>();
                    elsa.AddActivity<UpdateNotifyMail>();
                    elsa.AddActivity<QueryDataBaseActivity>();
                    elsa.AddActivity<GetScalarActivity>();
                    elsa.AddActivity<GetObjectApprovalInfoActivity>();
                });

                var apiOption = new Elsa.Server.Api.ElsaApiOptions();

                apiOption.SetupNewtonsoftJson = new Action<MvcNewtonsoftJsonOptions>(c => c.SerializerSettings.ContractResolver = new DefaultContractResolver());

                services.AddElsaApiEndpoints(apiOption);
                services.AddNotificationHandlers(typeof(ElsaJavaScriptExtender));

**Comment by sfmskywalker** [comment]
Thank you for sharing that. Unfortunately, there's nothing I see that's out of the ordinary. One thing you could maybe try is create a stripped down copy of your project with just the bare minimum?

**Comment by sfmskywalker** [comment]
Elsa 2.4.2 just got released to NuGet - perhaps worth a shot to try and upgrade to that?

**Comment by akghasemi** [comment]
ok,I will try new version

**Comment by akghasemi** [comment]
unfortunately the problem exist in latest version but I figured out something.
Obviously this "elsa-workflow-properties-panel" is causing the problem. what is that?
should it be set somewhere. because just right after this line it tries to render test panel and it hits error
![image](https://user-images.githubusercontent.com/46082076/150939182-bcac77bb-8bdd-43eb-877e-1c852244b00c.png)

**Comment by sfmskywalker** [comment]
The `elsa-workflow-properties-panel` is a custom element, like pretty much everything else. That's all taken care of by StencilJS, so you don't need to set anything yourself.

**Comment by akghasemi** [comment]
Hi,
I have found the reason for the error, in my startup file :
   

var apiOption = new Elsa.Server.Api.ElsaApiOptions();
  apiOption.SetupNewtonsoftJson = new Action<MvcNewtonsoftJsonOptions>(c => c.SerializerSettings.ContractResolver = new DefaultContractResolver());

 services.AddElsaApiEndpoints(apiOption);

this is causing problem when I comment the second line then designer loads without problem.
can you try to reproduce now

**Comment by akghasemi** [comment]
@sfmskywalker any updates?

**Comment by sfmskywalker** [comment]
I’m afraid not. It works on the sample server. If you send me a repro I can look at them I can try to look into it today.

**Comment by akghasemi** [comment]
[ElsaTest.zip](https://github.com/elsa-workflows/elsa-core/files/8013558/ElsaTest.zip)
@sfmskywalker here is the sample.
as I said earlier if you comment this line in startup
apiOption.SetupNewtonsoftJson = new Action(c => c.SerializerSettings.ContractResolver = new DefaultContractResolver());

then it works

**Comment by akghasemi** [comment]
@sfmskywalker did you have time to check the sample that I have provided?

**Comment by sfmskywalker** [comment]
I just checked and I was able to reproduce the issue.

The reason the designer breaks when providing custom JSON settings is that all of a sudden the API responses aren't what the designer expects anymore.

For example, the designer expects `camelCased` field names, so it will fail when receiving `PascalCased` fields.

I will push a fix that no longer relies on global MVC API options for these Elsa API endpoint controllers; instead, they will handle serializing responses manually.
